### PR TITLE
Validate default-cut-off-policy with gc config/repository APIs

### PR DESCRIPTION
**Pull request by nk1506**
config/repository is accepting string as defaultCutOffPolicy. which is very vague. 
Validating this using regex with Duration/ISO Instant/Positive Numeric values.

### CODE REVIEW

File: api/model/src/main/java/org/projectnessie/model/Validation.java [hunk 1]
```diff
@@ -95,6 +95,24 @@ public final class Validation {
   public static final Pattern REF_NAME_PATH_ELEMENT_PATTERN =
       Pattern.compile(REF_NAME_PATH_ELEMENT_REGEX);
 
+  /*
+   * Default Cut-Off Policy can be a number or Duration or ISO instant.
+   * Following rules are to validate the same.
+   * */
+  private static final String DURATION_REGEX =
```
Comment (ajantha-bhat) [inline api/model/src/main/java/org/projectnessie/model/Validation.java:102]: question: what about year and month?

**Review by snazy — APPROVED** [review]
Thanks!
LGTM!

File: api/model/src/main/java/org/projectnessie/model/Validation.java [hunk 2]
```diff
@@ -95,6 +95,24 @@ public final class Validation {
   public static final Pattern REF_NAME_PATH_ELEMENT_PATTERN =
       Pattern.compile(REF_NAME_PATH_ELEMENT_REGEX);
 
+  /*
+   * Default Cut-Off Policy can be a number or Duration or ISO instant.
```
Comment (ajantha-bhat) [inline api/model/src/main/java/org/projectnessie/model/Validation.java:99]: this comment may be applicable for `DEFAULT_CUT_OFF_POLICY_REGEX` not `DURATION_REGEX`

File: api/model/src/main/java/org/projectnessie/model/Validation.java [hunk 3]
```diff
@@ -196,6 +214,26 @@ public static String validateHash(String referenceName) throws IllegalArgumentEx
...
+    return matcher.matches();
+  }
+
+  /**
+   * Validate default cutoff policy.Policies can be one of: - number of commits as an integer value,
+   * - a duration (see java.time.Duration), - an ISO instant, - 'NONE,' means everything's
+   * considered as live
+   */
+  public static void validateDefaultCutOffPolicy(String value) {
+    if (isValidDefaultCutoffPolicy(value)) {
```
Comment (ajantha-bhat) [inline api/model/src/main/java/org/projectnessie/model/Validation.java:230]: I think we can inline `isValidDefaultCutoffPolicy` in this method and no need of private method.

**Comment by ajantha-bhat** [comment]
oh, I didn't refresh the tab while reviewing. It was a race condition :D

File: api/model/src/main/java/org/projectnessie/model/Validation.java (hunk 1 above)
Comment (ajantha-bhat) [reply api/model/src/main/java/org/projectnessie/model/Validation.java:102]: I might have confused with seeing example of string representation of `Period`. 
Duration doesn't have years and month
https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toString--

Comment (nk1506) [reply api/model/src/main/java/org/projectnessie/model/Validation.java:102]: We are using same format for nessie-gc . Also it seems Java Duration doesn't support others. 
The same pattern I took from Java class. 
`
 private static class Lazy {
        static final Pattern PATTERN =
            Pattern.compile("([-+]?)P(?:([-+]?[0-9]+)D)?" +
                    "(T(?:([-+]?[0-9]+)H)?(?:([-+]?[0-9]+)M)?(?:([-+]?[0-9]+)(?:[.,]([0-9]{0,9}))?S)?)?",
                    Pattern.CASE_INSENSITIVE);
    }
`

**Comment by ajantha-bhat** [comment]
Question is answered and other comments are nit. So, No need a follow up. 

LGTM

**Pull request merged**